### ISSUE TITLE
Openvpn client unable to adjust routes on Haiku

### ISSUE THREAD

**Issue by kallisti5** [OpenVPN/openvpn#458]
**Describe the bug**
OpenVPN is unable to adjust routes on Haiku and needs that knowledge added.

```
NOTE: unable to redirect IPv4 default gatgeway -- Cannot read current default gateway from system
```

**To Reproduce**
openvpn --config whatever.conf

**Expected behavior**
OpenVPN to be able to read and adjust routes on Haiku

**Version information (please complete the following information):**
 - OS: Haiku
 - OpenVPN version: 2.6.

**Additional context**
Haiku is about to receive proper tun device support, thus allowing openvpn to function :partying_face: 

I'm mostly opening this for tracking.  Someone from Haiku should submit a PR.

**Comment by kallisti5** [comment]
I completely  posted this too soon. I didn't realize our work wasn't upstream yet.  Closing for now :wink:

**Comment by cron2** [comment]
This does sound exciting :-) - so, for a complete port, you need to add a haiku-specific `get_default_gateway()` implementation (route.c), which will make automatic route redirection work.

When you send your patches, please rebase them on "master".  New features of larger scale are not merged in a release branch (like, 2.6.x).
